### PR TITLE
feat(login): base class, more DRY. Try to use more sane naming conven…

### PR DESCRIPTION
**Pull request by Avantol13**
### Improvements
- Base class for different IDPs for Login
- Update default configuration with details about setting up ORCID & Microsoft OAuth 2 Clients

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
The style in this PR agrees with [`black`](https://github.com/ambv/black). :heavy_check_mark:

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: fence/blueprints/login/base.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import flask
+from flask_restful import Resource
+
+from fence.auth import login_user
+from fence.config import config
+from fence.errors import UserError
+
+
+class DefaultOAuth2Login(Resource):
+    def __init__(self, idp_name, client, mock_username="test@example.com"):
```
Comment (rudyardrichter) [inline fence/blueprints/login/base.py]: It feels a little weird having the mock username as an option here, is there a case where just the config is not sufficient?

File: fence/blueprints/login/base.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        self.idp_name = idp_name
+        self.client = client
+        self.mock_username = mock_username
+
+    def get(self):
+        flask.redirect_url = flask.request.args.get("redirect")
+        if flask.redirect_url:
+            flask.session["redirect"] = flask.redirect_url
+
+        config_name = "MOCK_{}_AUTH".format(self.idp_name.upper())
```
Comment (rudyardrichter) [inline fence/blueprints/login/base.py]: using this format feels a little magical to me, what would you think about organizing it like this?
```yaml
OPENID_CONNECT:
  google:
    ...
    mock: true
    mock_username: 'testuser@gmail.com'
```
```python
if config.get(self.idp_name, {}).get("mock"):
    ...
```

File: fence/blueprints/login/base.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        if config.get(config_name, False):
+            username = flask.request.cookies.get(
+                config.get("DEV_LOGIN_COOKIE_NAME"), self.mock_username
+            )
+            return _login(username, self.idp_name)
+
+        return flask.redirect(self.client.get_auth_url())
+
+
+class DefaultOAuth2Callback(Resource):
```
Comment (rudyardrichter) [inline fence/blueprints/login/base.py]: :ok_hand:

File: fence/blueprints/login/base.py (hunk 2 above)
Comment (Avantol13) [reply fence/blueprints/login/base.py]: ooh, that's definitely more elegant. for backwards compatibility I think I need to at least still support the MOCK_GOOGLE_AUTH config :thinking: but I'll try to update this and let you know

Comment (Avantol13) [reply fence/blueprints/login/base.py]: actually... I think I have a neat solution for the backwards compatibility thing...

**Pull request merged**